### ISSUE TITLE
onKeyEnterEvent only works for the suggestion list, not the input itself

### ISSUE THREAD

**Issue by dvirsky** [pawelczak/EasyAutocomplete#195]
according to the documentation, the onKeyEnter event `Invokes function when user types "enter" in input field.`. 

That's not what happens - it invokes the function only if you hit enter on the suggestion list. 

So basically this leaves me with no way to know when a user wants the current text to be submitted to an API, and not just one of the suggestions.

The fix is simple and I did it myself, but it seems to me that a callback for the input should not be part of the list configuration, so I'm not submitting this as a pull request.

My suggestion is:
1. Change the documentation.
2. Add a new event listener for the input itself and not under the `list` configuration.

IMO there's no need for this event for the list itself, it's covered by other events.

**Comment by dmartin** [comment]
@dvirsky Would you mind including your fix for correctly passing along Enter key events from the input?

**Comment by amirkhan81** [comment]
It seems this still causes the selected element to return undefined.

**Comment by pacogvillaran** [comment]
yep, definitely this event does not work as said. For those looking for a quick fix, just rebind it like: searchBar.keydown(function(e){
				if (e.keyCode == 13)
					yourFunc();
			});

**Comment by vikinot** [comment]
Edit:
Moving the call to onKeyEnterEvent()  is fetch data on tab and enter keys both:

function bindKeydown() {
                 $field
                     .on("keydown", function(evt) {
                        evt = evt || window.event;
                        var keyCode = evt.keyCode;
                        if (keyCode === 38) {
                            suppressKeypress = true; 
                            return false;
                        }
                    })
                    .keydown(function(event) {
                        if ((event.keyCode === 13 || event.keyCode === 9) && selectedElement == -1 && elementsList.length > 0) {                            
                            selectedElement = 0;                           
                            $field.val(config.get("getValue")(elementsList[selectedElement]));
                            
                            config.get("list").onKeyEnterEvent();	
                            config.get("list").onChooseEvent();

                            selectedElement = -1;
			    hideContainer();
							
			  //event.preventDefault();   // handle the tab issue.
                        }
                        if (event.keyCode === 13 && selectedElement > -1) {
                            $field.val(config.get("getValue")(elementsList[selectedElement]));

                            config.get("list").onChooseEvent();

                            selectedElement = -1;
                            hideContainer();

                            event.preventDefault();
                        }
                    });
            }
